qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -642,9 +639,11 @@ def write(self, cmd):
         if you want to use this response, we put it in self._write_response
         (but only for the very last write call)
 
-        Note that this procedure makes it cumbersome to handle the returned
-        messages from concatenated commands, e.g. 'wav 1 1 1 0;fun 2 1 100 1 1'
-        Please don't use concatenated commands
+        In this method we expect to read one termination char per command. As
+        commands are concatenated by `;` we count the number of concatenated
+        commands as count(';') + 1 e.g. 'wav 1 1 1 0;fun 2 1 100 1 1' is two
+        commands. Note that only the response of the last command will be
+        available in `_write_response`
 
         TODO (WilliamHPNielsen): add automatic de-concatenation of commands.
```

Comment:
This PR just does that: it adds automatic de-concatenation (or at least something equivalent). This `TODO` should thus be removed.

Reply:
Done